language_detector/main.py

```diff
@@ -1,8 +1,65 @@
 # -*- coding: utf-8 -*-
 
 """This is the entry point of the program."""
+import re
+import collections
 
 def detect_language(text, languages):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    
+    # counter var created for each language, initiated with '0'
+    dictionar_of_languages = {}
+    for lang in languages:
+        dictionar_of_languages[lang['name']] = 0
+
+    # finds all words from string and strips the unnecessary symbols
+    list_of_words_in_text = re.findall(r'\w+',text)
+    
+    # access common words, and iterate over them
+    for word in list_of_words_in_text:
+        for lang in languages:
```

Comment:
Since you are iterating over languages two times (before in line 12) May be you can just iterate just once , create dictionary and proceed checking if the word exist in the common_words for that language. (Just a suggestion)
```python
dictionar_of_languages = {}
for lang in languages:
    dictionar_of_languages[lang['name']] = 0
    for word in list_of_words_in_text:
        if word in lang['common_words']:
            dictionar_of_languages[lang['name']]+=1
return dictionar_of_languages #and proceed
```